### ISSUE TITLE
acme.sh times out on CA processing for some but not all my domains

### ISSUE THREAD

**Issue by SeijiSensei** [acmesh-official/acme.sh#3882]
acme.sh works for some domains, fails for others. 

For some of my domains, e.g., takinganimeseriously.com, I first get this

[Mon Jan 10 19:40:09 UTC 2022] d='takinganimeseriously.com'
[Mon Jan 10 19:40:09 UTC 2022] ok, let's start to verify
[Mon Jan 10 19:40:09 UTC 2022] Verifying: takinganimeseriously.com
[Mon Jan 10 19:40:09 UTC 2022] d='takinganimeseriously.com'
[Mon Jan 10 19:40:09 UTC 2022] keyauthorization='-3JjKS2zTWmuHdDreYDmssPb1NsSAmklb9yShwRQRs8.oTKat2dbKLb52jdxuiHenG9Ci9vyticltjB0kcYvKHc'
[Mon Jan 10 19:40:09 UTC 2022] uri='https://acme.zerossl.com/v2/DV90/chall/D9kMxPAnZ4n8h-Cx_AgKjg'
[Mon Jan 10 19:40:09 UTC 2022] _currentRoot='apache'
[Mon Jan 10 19:40:10 UTC 2022] wellknown_path='/home/.acme'
[Mon Jan 10 19:40:10 UTC 2022] writing token:-3JjKS2zTWmuHdDreYDmssPb1NsSAmklb9yShwRQRs8 to /home/.acme/-3JjKS2zTWmuHdDreYDmssPb1NsSAmklb9yShwRQRs8
[Mon Jan 10 19:40:10 UTC 2022] url='https://acme.zerossl.com/v2/DV90/chall/D9kMxPAnZ4n8h-Cx_AgKjg'
[Mon Jan 10 19:40:10 UTC 2022] payload='{}'
[Mon Jan 10 19:40:10 UTC 2022] POST

then an repeated set of 

[Mon Jan 10 19:40:11 UTC 2022] Processing, The CA is processing your order, please just wait. (1/30)
[Mon Jan 10 19:40:11 UTC 2022] sleep 2 secs to verify again
[Mon Jan 10 19:40:13 UTC 2022] checking
[Mon Jan 10 19:40:13 UTC 2022] url='https://acme.zerossl.com/v2/DV90/chall/D9kMxPAnZ4n8h-Cx_AgKjg'
[Mon Jan 10 19:40:13 UTC 2022] payload
[Mon Jan 10 19:40:13 UTC 2022] POST
[Mon Jan 10 19:40:13 UTC 2022] _post_url='https://acme.zerossl.com/v2/DV90/chall/D9kMxPAnZ4n8h-Cx_AgKjg'
[Mon Jan 10 19:40:13 UTC 2022] _CURL='curl --silent --dump-header /root/.acme.sh/http.header  -L  -g '
[Mon Jan 10 19:40:15 UTC 2022] _ret='0'
[Mon Jan 10 19:40:15 UTC 2022] code='200'
[Mon Jan 10 19:40:15 UTC 2022] Processing, The CA is processing your order, please just wait. (2/30)
[Mon Jan 10 19:40:15 UTC 2022] sleep 2 secs to verify again

until it reaches 30 retries and times out.  

For other domains I own, the process works flawlessly. Is there something about the DNS records for this domain that might be causing the problem? Most of my domains have nearly identical DNS zone files, so I'm puzzled if that's the reason.

**Comment by SeijiSensei** [comment]
OK. I removed a TXT record from the zone file for takinganimeseriouusly.com and the request went through correctly. It was a "google-site-verification" record. I don't know whether the problem lay with acme.sh or the CA, but obviously this is a bug that needs fixing.

**Comment by sapinva** [comment]
Having the same issue starting last week. No errors with --debug 2, just times out. DNS records are fine. These are with the same domains that have renewed before with no problem.

**Comment by nicbou** [comment]
I have the same issue, but only with one domain. Both servers work exactly the same, and use the same code to issue certificates, but one consistently timeout. The logs look exactly like in the description above. I have no explanation.

**Comment by nicbou** [comment]
Updating my last comment: it looks like confusing messages and not an actual timeout. In my case, the .well_known endpoint was not accessible. The error message sent me in the wrong direction. Once I fixed the underlying issue, the validation was very quick.

It would not hurt to improve the log message, although my understanding is that each SSL provider reports errors differently. I only got a useful error message with Let's Encrypt.

The underlying issue was an invalid self-signed certificate preventing nginx from starting. The .well_known endpoint was unavailable and ZeroSSL couldn't validate that I really controlled that domain. 

For reference, this is the script I use to generate SSL certificates: https://github.com/nicbou/homeserver/blob/master/proxy/scripts/start.sh

tl;dr: **If you get this error, it could be that your .well_known validation endpoint is not accessible**

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Not a bug: reporters traced it to their own validation setup (conflicting TXT record, inaccessible .well-known); ZeroSSL just kept the order in processing.
